cmd/gen-config.go

```diff
@@ -0,0 +1,127 @@
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Azure/azure-storage-fuse/v2/common"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
+	"github.com/spf13/cobra"
+)
+
+type generatedConfigOptions struct {
+	configComp     string
+	configTmp      string
+	configDirectIO bool
+}
+
+var opts2 generatedConfigOptions
+
+var generatedConfig = &cobra.Command{
+	Use:               "gen-config",
+	Short:             "Generate default config file.",
+	Long:              "Generate default config file with the values pre-caculated by blobfuse2.",
+	SuggestFor:        []string{"generate default config", "generate config"},
+	Hidden:            true,
+	Args:              cobra.ExactArgs(0),
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		common.GenConfig = true
+
+		if opts2.configComp != "block_cache" && opts2.configComp != "file_cache" {
+			return fmt.Errorf("component is required and should be either block_cache or file_cache")
+		}
+
+		// Check if configTmp is not provided when component is fc
+		if opts2.configComp == "file_cache" && opts2.configTmp == "" {
+			return fmt.Errorf("temp path is required for file cache mode. Use flag --tmp-path to provide the path")
+		}
+
+		pipeline := []string{"libfuse"}
+		if opts2.configComp == "block_cache" {
+			pipeline = append(pipeline, "block_cache")
+			common.TmpPath = opts2.configTmp
+		} else if opts2.configComp == "file_cache" {
+			pipeline = append(pipeline, "file_cache")
+			common.TmpPath = opts2.configTmp
+		}
+		if opts2.configDirectIO {
+			common.DirectIO = true
+		} else {
+			pipeline = append(pipeline, "attr_cache")
+		}
+
+		yamlContent := "# Logger configuration\n#logging:\n  #  type: syslog|silent|base\n  #  level: log_off|log_crit|log_err|log_warning|log_info|log_trace|log_debug\n  #  file-path: <path where log files shall be stored. Default - '$HOME/.blobfuse2/blobfuse2.log'>\n"
```

Comment:
Let's replace yamlContent from string to stringbuilder.
Let's also make it in multi-line for easy understanding.
Sample -

var sb strings.Builder
		sb.WriteString("# Logger configuration\n#logging:\n")
		sb.WriteString("  #  type: syslog|silent|base\n")
		sb.WriteString("  #  level: log_off|log_crit|log_err|log_warning|log_info|log_trace|log_debug\n")
		sb.WriteString("  #  file-path: <path where log files shall be stored. Default - '$HOME/.blobfuse2/blobfuse2.log'>\n")